lib/child_process.js

```diff
@@ -196,7 +196,12 @@ function normalizeExecArgs(command, options, callback) {
 
   // Make a shallow copy so we don't clobber the user's options object.
   options = { __proto__: null, ...options };
-  options.shell = typeof options.shell === 'string' ? options.shell : true;
+
+  // Validate the shell, if present, and ensure a truthy value.
+  if (options.shell != null) {
+    validateString(options.shell, 'options.shell');
+  }
+  options.shell ||= true;
```

Comment:
I don't understand why would we want to accept the empty string a valid value
```suggestion
  } else {
    options.shell = true;
  }
```

Reply:
The sister functions spawn/execFile validate `shell: ''` as valid, and their behaviour then becomes the same as passing `false` or `undefined`.

This PR makes exec's behaviour the same as these, but the empty string needs to be coerced to `true` just the same as nullish values, otherwise a shell won't be spawned (as per the logic outlined in the previous PR's description).

Reply:
This suggested change would maintain the existing edge case behaviour when `{ shell: '' }` is passed to exec.

- It is not null, and validates as a string, so is passed through unchanged.
- When this reaches `spawn()` downstream, because it is not a truthy value, a shell is _not_ invoked.

`exec()` must _always_ invoke a shell, and therefore anything that passes this validation needs to be a truthy value.

Reply:
It would make more sense to throw on empty string IMO

Reply:
I'd agree, but this would then diverge from the behaviour of the other spawning child_process functions, unless we change them all?

Reply:
I guess we could start by deprecating that, sure

Reply:
Separate PR?

Reply:
Yes, two separate PRs eve, one for doc-only deprecation, and one that adds a runtime warning. Would you like to take up that effort?

Reply:
IMO it should be the current behavior until we move forward with the deprecation at which point it will throw. I don't see the point of introducing an intermediary breaking change only to land another breaking change